managed/ui/src/components/config/Security/ListKeyManagementConfigurations.js

```diff
@@ -1,28 +1,48 @@
 // Copyright (c) YugaByte, Inc.
 
 import React, { Component, Fragment } from 'react';
-import { Button } from 'react-bootstrap';
+import { Button, DropdownButton, MenuItem } from 'react-bootstrap';
 import { FlexContainer, FlexShrink } from '../../common/flexbox/YBFlexBox';
 import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
 import { YBPanelItem } from '../../panels';
+import AssociatedUniverse from '../../common/associatedUniverse/AssociatedUniverse';
 
 export class ListKeyManagementConfigurations extends Component {
-  render() {
-    const { configs, onCreate, onDelete } = this.props;
+  state = {
+    associatedUniverses: []
+  }
 
-    const actionList = (item, row) => {
-      const { configUUID, in_use } = row.metadata;
-      return (
-        <Button
+  actionList = (item, row) => {
+    const { configUUID, in_use, universeDetails } = row.metadata;
+    return (
+      <DropdownButton className="btn btn-default" title="Actions" id="bg-nested-dropdown" pullRight>
+        <MenuItem
           title={'Delete provider'}
-          bsClass="btn btn-default btn-config pull-right"
           disabled={in_use}
-          onClick={() => onDelete(configUUID)}
+          onClick={() => this.props.onDelete(configUUID)}
         >
-          Delete Configuration
-        </Button>
-      );
-    };
+          <i className="fa fa-trash"></i> Delete Configuration
+        </MenuItem>
+        <MenuItem
+          onClick={() => {
+            this.setState({ associatedUniverses: [...universeDetails] });
+            this.props.showassociatedUniversesModal();
```

Comment:
you are triggering `AssociatedUniverse` modal via redux action but providing it universes list via local state
let's stick to the uniform approach pls - either keep `visibleModal` flag in local state or provide universes list for the modal via redux store

Reply:
I am sticking with the local state as we need to pass the universe list for the selected row, Thus it is easier if we pass it locally